File: frontend/src/types/equipment.ts
export interface Equipment {
    id: number;
    name: string;
    category: EquipmentCategory;
    icon: string;
    brand: string;
    model: string;
    serialNumber?: string;
    purchaseDate: string;
    purchaseCost: number;
    warrantyExpiryDate?: string;
    vendorName?: string;
    quantity: number;
    location: string;
    status: EquipmentStatus;
    condition: EquipmentCondition;
    usageLevel?: UsageLevel;
    lastMaintenanceDate?: string;
    nextMaintenanceDueDate?: string;
    createdAt?: string;
    updatedAt?: string;
}

export type EquipmentCategory = 'STRENGTH' | 'CARDIO' | 'FUNCTIONAL' | 'YOGA' | 'RECOVERY' | 'OTHER';
export type EquipmentStatus = 'ACTIVE' | 'MAINTENANCE' | 'OUT_OF_ORDER' | 'RETIRED';
export type EquipmentCondition = 'NEW' | 'GOOD' | 'FAIR' | 'POOR';
export type UsageLevel = 'LOW' | 'MEDIUM' | 'HIGH';

export interface EquipmentStats {
    total: number;
    active: number;
    maintenance: number;
    outOfOrder: number;
}

File: backend/src/main/java/com/gym/management/model/Equipment.java
package com.gym.management.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "equipment")
public class Equipment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "gym_id")
    private Long gymId;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EquipmentCategory category;

    private String icon; // Iconscout icon name

    private String brand;
    private String model;
    private String serialNumber;

    private LocalDate purchaseDate;
    private BigDecimal purchaseCost;
    private LocalDate warrantyExpiryDate;
    private String vendorName;

    private Integer quantity;
    private String location;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EquipmentStatus status;

    @Enumerated(EnumType.STRING)
    @Column(name = "equipment_condition")
    private EquipmentCondition condition;

    @Enumerated(EnumType.STRING)
    private UsageLevel usageLevel;

    private LocalDate lastMaintenanceDate;
    private LocalDate nextMaintenanceDueDate;

    private boolean isDeleted = false;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public enum EquipmentCategory {
        STRENGTH, CARDIO, FUNCTIONAL, YOGA, RECOVERY, OTHER
    }

    public enum EquipmentStatus {
        ACTIVE, MAINTENANCE, OUT_OF_ORDER, RETIRED
    }

    public enum EquipmentCondition {
        NEW, GOOD, FAIR, POOR
    }

    public enum UsageLevel {
        LOW, MEDIUM, HIGH
    }
}
